3rdparty/mbedtls/mbedtls_hardware_poll.c

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/entropy.h>
+
+/*
+ * MBEDTLS links this function definition when MBEDTLS_ENTROPY_HARDWARE_ALT
+ * is defined in the MBEDTLS config.h file. This is the sole source of entropy
+ * for MBEDTLS. All other MBEDTLS entropy sources are disabled since they don't
+ * work within enclaves.
+ */
+int mbedtls_hardware_poll(
+    void* data,
+    unsigned char* output,
+    size_t len,
+    size_t* olen)
+{
+    int ret = -1;
```

Comment:
Can you follow up with @mikbras if the convention is to always initialize out parameters to known values, even in case of failures?
If so, in this case, we would initialize *olen to 0, for a non null olen.

Reply:
@mikbras can you comment on this?